osm2streets/src/lanes/mod.rs

```diff
@@ -355,3 +397,81 @@ impl fmt::Display for Direction {
         }
     }
 }
+
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, PartialOrd, Ord, Serialize, Deserialize)]
+pub enum LtrLaneNum {
+    Forward(usize),
+    Backward(usize),
+}
+
+impl LtrLaneNum {
+    pub fn direction(&self) -> Direction {
+        match self {
+            Self::Forward(_) => Direction::Fwd,
+            Self::Backward(_) => Direction::Back,
+        }
+    }
+
+    pub fn number(&self) -> usize {
+        match self {
+            Self::Forward(num) | Self::Backward(num) => *num,
+        }
+    }
+
+    /// Converts to the same numbered lane in the opposite direction.
+    pub fn reverse(&self) -> Self {
+        use LtrLaneNum::*;
+        match self {
+            Forward(n) => Backward(*n),
+            Backward(n) => Forward(*n),
+        }
+    }
+}
+
+/// Identifies a position within the width of a roadway. Lanes are identified by their left-to-right
+/// position, as per the OSM convention.
+///
+/// Most commonly seen as a value of the placement tag, e.g.
+/// `placement=right_of:1` means that the OSM way is drawn along the right edge of lane 1.
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, PartialOrd, Ord, Serialize, Deserialize)]
+pub enum RoadPosition {
+    /// The center of the carriageway width, ignoring lanes. The default placement of OSM ways.
+    Center,
```

Comment:
Seems awkward that this type uses "center" and "middle" to mean the same thing, but "center" matches with the convention in this repo, and "middle" matches the placement tag. Opinion?